packages/react-native/src/index.ts

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable global-require */
+
+import 'regenerator-runtime/runtime';
+
+import { createSDK } from '@magic-sdk/provider';
+import { ReactNativeWebViewController } from './react-native-webview-controller';
+import { ReactNativeTransport } from './react-native-transport';
+import { SDKBaseReactNative } from './react-native-sdk-base';
+
+// We expect `global.process` to be a Node Process, so we have to replace it
+// here.
+global.process = require('process');
+
+// WHATWG URL requires global `Buffer` access.
+/* istanbul ignore next */
+// We cannot test this code path without causing problems for Ava & TS Node.
+// Unfortunatley, we must ignore it.
+if (typeof global.Buffer === 'undefined') global.Buffer = require('buffer').Buffer;
+
+(process as any).browser = false;
+
+// Web3 assumes a browser context, so we need to provide a `btoa` shim.
+if (typeof btoa === 'undefined') {
+  /* istanbul ignore next */
+  global.btoa = str => {
+    return Buffer.from(str, 'binary').toString('base64');
+  };
+}
+
+// Web3 assumes a browser context, so we need to provide an `atob` shim.
+if (typeof atob === 'undefined') {
+  /* istanbul ignore next */
+  global.atob = b64Encoded => {
+    return Buffer.from(b64Encoded, 'base64').toString('binary');
+  };
+}
+
+/* istanbul ignore next */
+// We cannot test this code path without causing problems for Ava & TS Node.
+// Unfortunatley, we must ignore it.
+if (typeof URL === 'undefined') {
```

Comment:
I get URL “Not implemented” error while testing with the canary version. https://github.com/facebook/react-native/issues/25187
 
I think it's due to the backward compatibility has been removed here. https://github.com/magiclabs/magic-js/blob/monorepo/packages/provider/src/util/url.ts

Reply:
You can test it on their snack without further configuration on your local

https://snack.expo.io/@jerry-liu/magic-link-test

Reply:
I’ll take a look today, see if I can narrow down the cause

Reply:
I think I've been able to resolve this in `@magic-sdk/react-native@1.0.0-canary.1`.